ql/math/beta.cpp

```diff
@@ -72,10 +72,8 @@ namespace QuantLib {
     QL_REQUIRE(b > 0.0, "b must be greater than zero");
 
 
-    if (x == 0.0)
-        return 0.0;
-    else if (x == 1.0)
-        return 1.0;
+    if (x == 0.0 || x == 1.0)
+        return x;
```

Comment:
Thanks!  I think the old version might read clearer, though — "here is the limit on the left boundary, and here's the limit on the right one".  The fact that the two clauses can be joined is kind of accidental, I guess?

Reply:
Yeah , I agree with it that old version read clearer.. But rather than complicate by more if - else if condition it's better to look and somewhat better to read.